tests/issues.rs

```diff
@@ -105,3 +105,51 @@ mod issue514 {
         assert_eq!(reader.read_event().unwrap(), Event::Eof);
     }
 }
+
+#[test]
+fn issue_604_short() {
+    let data = b"<?xml version=\"1.0\" encoding=\"utf-8\"?><!-->";
+    let mut reader = Reader::from_reader(data.as_slice());
+    let mut buf = Vec::new();
+    assert!(matches!(
+        reader.read_event_into(&mut buf),
+        Ok(Event::Decl(_))
+    ));
```

Comment:
Use `assert_eq!()` + `unwrap()` for `Ok` results, like as above. `assert_eq!()` is from `pretty_assertions` crate which will print a nice diff in case of error